src/base/rss/rssdownloadrule.cpp

```diff
@@ -50,37 +50,101 @@ DownloadRule::DownloadRule()
 
 bool DownloadRule::matches(const QString &articleTitle) const
 {
+    qDebug() << "Matching article:" << articleTitle;
+
+    qDebug() << "Checking matching tokens:" << m_mustContain.join("|");
+    bool foundMustContain = false;
+
     foreach (const QString &token, m_mustContain) {
-        if (!token.isEmpty()) {
-            QRegExp reg(token, Qt::CaseInsensitive, m_useRegex ? QRegExp::RegExp : QRegExp::Wildcard);
-            if (reg.indexIn(articleTitle) < 0)
-                return false;
+        if (token.isEmpty()) {
+            continue;
+        }
+
+        if (m_useRegex) {
+            QRegExp reg(token, Qt::CaseInsensitive, QRegExp::RegExp);
+
+            if (reg.indexIn(articleTitle) > -1) {
+                foundMustContain = true;
+            }
+        }
+        else {
+            // Split the token on spaces, then check if all of the elements are found (in any order) - if so, it's a potential match.
+            foundMustContain = true;
+
+            // We need to call replace, so can't be const &
+            foreach (QString wildToken, token.split(" ")) {
+                QRegExp reg(wildToken.replace(" ", ""), Qt::CaseInsensitive, QRegExp::Wildcard);
+
+                if (reg.indexIn(articleTitle) == -1) {
+                    foundMustContain = false;
+                    break;
+                }
+            }
         }
+
+        if (foundMustContain) {
+            qDebug() << "Found matching token: " << token;
+            break;
+        }
+    }
+
+    if (!foundMustContain) {
+        return false;
     }
-    qDebug("Checking not matching tokens");
-    // Checking not matching
+
+    qDebug() << "Checking not matching tokens:" << m_mustNotContain.join("|");
+
     foreach (const QString &token, m_mustNotContain) {
-        if (!token.isEmpty()) {
-            QRegExp reg(token, Qt::CaseInsensitive, m_useRegex ? QRegExp::RegExp : QRegExp::Wildcard);
-            if (reg.indexIn(articleTitle) > -1)
+        if (token.isEmpty()) {
+            continue;
+        }
+
+        if (m_useRegex) {
+            QRegExp reg(token, Qt::CaseInsensitive, QRegExp::RegExp);
+
+            if (reg.indexIn(articleTitle) > -1) {
+                qDebug() << "Found not matching token: " << token;
                 return false;
+            }
+        }
+
+        // Split the token on spaces, then check if all of the elements are found (in any order) - if so, reject it.
```

Comment:
I think you've misunderstood the code. For each whitespace-delimited `wildToken`, if it is **not** present in the article title (`indexIn() == -1`) then we record that we didn't find a match. We know that there's at least one `wildToken`. So for `foundMustNotContain` to still be `true` at the end of the loop, none of the whitespace-delimited tokens must have been present in the article name.